compiler/rustc_passes/src/check_attr.rs

```diff
@@ -530,6 +540,11 @@ fn is_c_like_enum(item: &Item<'_>) -> bool {
 fn check_mod_attrs(tcx: TyCtxt<'_>, module_def_id: LocalDefId) {
     tcx.hir()
         .visit_item_likes_in_module(module_def_id, &mut CheckAttrVisitor { tcx }.as_deep_visitor());
+    if tcx.hir().local_def_id(CRATE_HIR_ID) == module_def_id {
+        for attr in tcx.hir().krate_attrs() {
+            CheckAttrVisitor { tcx }.check_doc_alias(attr, CRATE_HIR_ID, Target::Mod);
+        }
```

Comment:
We could call [`check_attributes`](https://github.com/rust-lang/rust/blob/07ece44a42b6cd33e59d58b5325f637f398464d2/compiler/rustc_passes/src/check_attr.rs#L52) here which should fix #64734 and #69365 while we're at it.

If you want to just apply this to the `#[doc]` attribute for now though this will need a `if self.tcx.sess.check_name(attr, sym::doc)` check.